Maple2.Model/Metadata/Table/MasteryRecipeTable.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using Maple2.Model.Enum;
+
+namespace Maple2.Model.Metadata;
+
+public record MasteryRecipeTable(IReadOnlyDictionary<int, MasteryRecipeTable.Entry> Entries) : Table {
+    public record Entry(int Id,
+                        MasteryType Type,
+                        bool NoRewardExp,
+                        int RequiredMastery,
+                        long RequiredMeso,
+                        IReadOnlyList<int> RequiredQuests,
+                        long RewardExp,
+                        int RewardMastery,
+                        int HighPropLimitCount,
+                        int NormalPropLimitCount,
+                        IReadOnlyList<Ingredient> RequiredItems,
+                        IReadOnlyList<int> HabitatMapId,
+                        IReadOnlyList<Ingredient> RewardItems);
+
+    public record Ingredient(int ItemId, short Rarity, int Amount, string Tag);
```

Comment:
I will push my change, and you can use the `ItemTag` enum here instead of string. It's weird that they have both ItemId and Tag, perhaps itemId is useless?

Reply:
itemId is useless, yeah. if a tag is present, it uses that. I suppose itemId it's really just for display purposes.

Reply:
I think we can skip parsing it then. I wonder if `rarity` is the unknown field in [IngredientInfo](https://github.com/kOchirasu/Maple2/blob/master/Maple2.Model/Game/Item/IngredientInfo.cs)

Reply:
In case of mastery recipes, we should still parse it. There are some ingredients that _dont_ use ItemTags.
What uses IngredientInfo?